src/graph/ShowExecutor.cpp

```diff
@@ -967,6 +970,184 @@ void ShowExecutor::showCollation() {
+        std::vector<cpp2::RowValue> rows;
+        std::vector<std::string> header{"Account",
+                                        "IsLock",
+                                        "Login Type",
+                                        "MAX_QUERIES_PER_HOUR",
+                                        "MAX_UPDATES_PER_HOUR",
+                                        "MAX_CONNECTIONS_PER_HOUR",
+                                        "MAX_USER_CONNECTIONS"};
+        resp_->set_column_names(std::move(header));
+        std::vector<cpp2::ColumnValue> row;
+        row.resize(7);
+        row[0].set_str(user.get_account());
+        row[1].set_bool_val(user.__isset.is_lock ? *user.get_is_lock() : false);
+        if (user.__isset.login_type) {
+            row[2].set_str(*user.get_login_type() == nebula::cpp2::UserLoginType::PASSWORD
+                           ? "PASSWORD" : "LDAP");
+        } else {
+            row[2].set_str("UNKNOWN");
+        }
+        row[3].set_integer(user.__isset.max_queries_per_hour ?
+                           *user.get_max_queries_per_hour() : 0);
+        row[4].set_integer(user.__isset.max_updates_per_hour ?
+                           *user.get_max_updates_per_hour() : 0);
+        row[5].set_integer(user.__isset.max_connections_per_hour ?
+                           *user.get_max_connections_per_hour() : 0);
+        row[6].set_integer(user.__isset.max_user_connections ?
+                           *user.get_max_user_connections() : 0);
+        rows.emplace_back();
+        rows.back().set_columns(std::move(row));
+
+        resp_->set_rows(std::move(rows));
+        doFinish(Executor::ProcessControl::kNext);
+    };
+
+    auto error = [this] (auto &&e) {
+        LOG(ERROR) << "Exception caught: " << e.what();
+        doError(Status::Error(folly::stringPrintf("Show tags exception: %s",
+                                                  e.what().c_str())));
+        return;
+    };
+    std::move(future).via(runner).thenValue(cb).thenError(error);
+}
+
+void ShowExecutor::showUsers() {
+    auto future = ectx()->getMetaClient()->listUsers();
+    auto *runner = ectx()->rctx()->runner();
+
+    auto cb = [this] (auto &&resp) {
+        if (!resp.ok()) {
+            doError(std::forward<decltype(resp)>(resp).status());
+            return;
+        }
+
+        auto value = std::forward<decltype(resp)>(resp).value();
+        resp_ = std::make_unique<cpp2::ExecutionResponse>();
+        std::vector<cpp2::RowValue> rows;
+        std::vector<std::string> header{"Account", "IsLock", "Login Type",
+                                        "MAX_QUERIES_PER_HOUR",
+                                        "MAX_UPDATES_PER_HOUR",
+                                        "MAX_CONNECTIONS_PER_HOUR",
+                                        "MAX_USER_CONNECTIONS"};
+        resp_->set_column_names(std::move(header));
+        for (auto& item : value) {
+            auto user = item.second;
```

Comment:
It seems we could unify code with `showUser`?

Reply:
good idea. I've unified.